internal/powerbi/resource_pbix.go

```diff
@@ -46,22 +46,26 @@ func ResourcePBIX() *schema.Resource {
 			},
 			"skip_report": {
 				Type:        schema.TypeBool,
-				Description: "If true only the PBIX dataset is deployed.",
+				Description: "If true, only the PBIX dataset is deployed.",
 				Optional:    true,
 				Default:     false,
 			},
 			"report_id": {
 				Type:        schema.TypeString,
 				Description: "The ID for the report that was deployed as part of the PBIX.",
-				Optional:    true,
```

Comment:
What does it mean that it is not `Optional` anymore? If `skip_report` is `true`, then there is no `report_id`, so I think it is considered Optional? Or does that just mean that its not settable from configuration then, in which case I think its fine.

Reply:
> If skip_report is true, then there is no report_id, so I think it is considered Optional

That was original thinking which is why it was optional. But after doing some futher reading it actually means that you can not define report_id (or dataset_id) in your terraform, it is entirely generated and you cant do anything other take the generated value